tools/code_analyzer/op_dependency.cpp

```diff
@@ -126,6 +126,15 @@ cl::opt<RegexOpt, true, cl::parser<std::string>> OpInvocationPattern(
     cl::Required,
     cl::ValueRequired);
 
+RegexOpt RootSymbolPatternLoc;
+cl::opt<RegexOpt, true, cl::parser<std::string>> RootSymbolPattern(
+    "root_symbol_pattern",
+    cl::desc("Regular expression used to identify root symbols. It will insert "
+             "an entry to the output graph with key = `__ROOT__` and value = "
+             "set of ops reachable from root symbols, if the pattern is set. "
+             "Example: -root_symbol_pattern 'torch::jit'"),
```

Comment:
I think it's still not altogether clear to me what I, as a user, should put here. At a guess, it sounds like a C++ user of the analysis tool should specify a root symbol to be some "reasonably initial" symbol, from which we take the transitive (static?) closure to look for all operator calls. So as a C++ user should I put "main" as the root symbol? But the example here suggests maybe I should be namespacing my C++ API calls in such a way that I can just match for that namespace? What's the idea?

Reply:
I added answers to these questions to the code as comment:
```
// Why not dump ops that are reachable from any visible C++ symbols? Why
// limit it to a subset of root symbols?
// Because op registration callsites in static initializer are visible root
// symbols, too. It will dump ALL the registered ops without any filtering.
//
// Can we use some fixed entry point like `main()`?
// The target to be analyzed can be DSO that doesn't have a `main()`.
```

"torch::jit" is a simple pattern I use to cover most visible torch APIs that could be called externally (might not actually be called in the binary that is being analyzed).

We can refine the pattern later on with more accurate set of symbols. It's only exposed to be called by the checked-in shell script - I'm not sure whether we want to expose this flag to external usage, yet.

Won't we need multiple root symbol patterns in the general case? Yes, I guess we can concatenate the symbols into one RE with "|"?